packages/api/endpoints/rules.js

```diff
@@ -167,6 +168,9 @@ async function put(req, res) {
       translatedRule = await translatePostgresRuleToApiRule(pgRule, knex);
       await indexRule(esClient, translatedRule, process.env.ES_INDEX);
     });
+
+    await deleteOldEventSourceMappings(knex, oldApiRule);
+
```

Comment:
There's an added issue here that catching error here *does not fix the updated triggers*

I do not believe this is an issue in main, I've issued CUMULUS-3206 against the feature branch to mitigate.